src/ja/server/web/requests.py

```diff
@@ -0,0 +1,111 @@
+from abc import ABC, abstractmethod
+from ja.common.message.base import Serializable
+from ja.server.database.database import ServerDatabase
+from typing import Dict
+
+
+class WebRequest(Serializable, ABC):
+    """
+    A base class for the different statistics requests.
+    """
+
+    @abstractmethod
+    def generate_report(self, database: ServerDatabase) -> None:
+        """!
+        Generate the requested statistics from the data in the database.
+        """
+
+
+class WorkMachineWorkloadRequest(WebRequest):
+    """
+    Generates the response to the machine workload request.
+    """
+
+    def __init__(self, uid: str):
+        """!
+        Initialize the request response.
+
+        @param uid The uid of the work machine this request is for.
```

Comment:
Can this be None to get the workload of all machines?

Reply:
I don't think so.

Reply:
You can use the workload request to get the IDs of all machines, and then you can issue a request for each machine to get the running jobs, if that's what the user needs.